test/powershell/Modules/Microsoft.PowerShell.Utility/Select-Object.Tests.ps1

```diff
@@ -362,13 +340,65 @@ Describe "Select-Object with Property = '*'" -Tags "CI" {
         $obj.psobject.TypeNames.Count | Should -Be 3
         $obj.psobject.TypeNames[0] | Should -BeLike "Selected*"
         $obj.psobject.TypeNames[1] | Should -Not -BeLike "Selected*"
+        $p = Get-Process -Id $pid | Select-Object -Property Process* -ExcludeProperty ProcessorAffinity -ExpandProperty Modules
+        $p[0].psobject.Properties.Item("ProcessorAffinity") | Should -BeNullOrEmpty
         $obj.psobject.TypeNames[2] | Should -Not -BeLike "Selected*"
+    }
+}
 
-        $obj = $obj | Select-Object
+Describe 'Select-Object behaviour with hashtable entries and actual members' -Tags CI {
 
-        $obj.psobject.TypeNames.Count | Should -Be 3
-        $obj.psobject.TypeNames[0] | Should -BeLike "Selected*"
-        $obj.psobject.TypeNames[1] | Should -Not -BeLike "Selected*"
-        $obj.psobject.TypeNames[2] | Should -Not -BeLike "Selected*"
+    It 'can retrieve a hashtable entry as a property' {
+        $hashtable = @{ Entry = 100 }
+
+        $result = $hashtable | Select-Object -Property Entry
+        $result.Entry | Should -Be 100
+
+        $hashtable |
+            Select-Object -ExpandProperty Entry |
+            Should -Be 100
+    }
+
+    It 'can retrieve true hashtable members' {
+        $hashtable = @{ Value = 10 }
+        $result = $hashtable | Select-Object -Property Keys
+        $result.Keys | Should -Be 'Value'
+
+        $hashtable |
+            Select-Object -ExpandProperty Keys |
+            Should -Be 'Value'
+    }
+
+    It 'should prioritise hashtable entries where available' {
+        $hashtable = @{ Keys = 10 }
+        $result = $hashtable | Select-Object -Property Keys
+        $result.Keys | Should -Be 10
+
+        $hashtable | Select-Object -ExpandProperty Keys | Should -Be 10
+    }
+
+    It 'should get the hashtable Count member' {
+        $hashtable = @{ a = 10; b = 20; c = 30 }
+        $result = $hashtable | Select-Object -Property Count
+        $result.Count | Should -Be 3
+
+        $hashtable | Select-Object -ExpandProperty Count | Should -Be 3
+    }
+
+    It 'should get the hashtable Count member' {
+        $hashtable = @{ a = 10; b = 20; c = 30 }
+        $result = $hashtable | Select-Object -Property Count
+        $result.Count | Should -Be 3
+
+        $hashtable | Select-Object -ExpandProperty Count | Should -Be 3
+    }
+
+    It 'should get the Length member from a hashtable' {
+        $hashtable = @{ a = 10; b = 20; c = 30 }
+
+        $result = $hashtable | Select-Object -Property Length
```

Comment:
> Can you please add a test on selecting the Count and Length properties of a hashtable?

Sorry about the `Length` part in my comment, I thought there was a `Length` alias defined for `Count` but I was wrong.
You can remove this test.

Reply:
@daxian-dbw thanks for clarifying! I thought there was for a moment as well, but seems not.